Ziti.NET.Standard/src/OpenZiti/ZitiIdentity.cs

```diff
@@ -295,10 +330,38 @@ public struct InitOptions {
 			internal void ZitiServiceEvent(ZitiServiceEvent evt) {
 				OnZitiServiceEvent?.Invoke(this, evt);
 			}
-		}
 
-		private void native_ziti_pq_mac_cb(IntPtr ziti_context, string id, Native.ziti_pr_mac_cb response_cb) {
+			internal void ZitiMFAEvent(ZitiMFAEvent evt)
+			{
+				OnZitiMFAEvent?.Invoke(this, evt);
+			}
+
+			internal void ZitiAPIEvent(ZitiAPIEvent evt)
+			{
+				OnZitiAPIEvent?.Invoke(this, evt);
+			}
+
+			internal void ZitiMFAStatusEvent(ZitiMFAStatusEvent evt)
+			{
+				OnZitiMFAStatusEvent?.Invoke(this, evt);
+			}
+		}
+		public struct TunnelCB
+		{
+			public ZitiIdentity.InitOptions zidOpts;
+			public delegate void ZitiResponseDelegate(object evt);
+
+			public void ZitiResponse(object evt)
+			{
+				if (evt is ZitiMFAStatusEvent)
+				{
+					zidOpts.ZitiMFAStatusEvent((ZitiMFAStatusEvent)evt);
+				}
+			}
+		}
 
+		public void native_ziti_pq_mac_cb(IntPtr ziti_context, string id, Native.ziti_pr_mac_cb response_cb) {

```

Comment:
`native_ziti_pq_mac_cb` doesn't belong on this class. if this is necessary it needs to be converted to idiomatic csharp.

Reply:
sure. it was most likely like committed as an afterthought by me. something i stubbed out to be 'fixed' later. we should either remove it entirely or 'fix' it is all i am trying to say. looking at the struct there are 3 other callbacks that would need the same kind of treatment. they seem to go unused though? Or maybe i just don't see the other callbacks used?